src/main/java/marquez/service/models/JobMeta.java

```diff
@@ -74,7 +78,7 @@ public UUID version(@NonNull NamespaceName namespaceName, @NonNull JobName jobNa
                     .map(output -> output.getValue())
                     .collect(joining(VERSION_DELIM)),
                 KV_JOINER.join(getContext()),
-                location.toString())
+                getLocation().map(location -> location.toString()).orElse(null))
```

Comment:
you could do `URL::toString` here

Reply:
Yes! And hand no idea this was an option 💯